src/retryOnEmpty.ts

```diff
@@ -140,7 +141,10 @@ async function retry(
   for (let index = 0; index < maxRetries; index++) {
     try {
       return await asyncFn();
-    } catch (err) {
+    } catch (err: any) {
+      if (isExecutionRevertedError(err)) {
+        throw err as unknown;
+      }
```

Comment:
Instead of asserting that `err` is `unknown`, can you tell TypeScript to use `unknown` as the type of `err` to begin with?

``` suggestion
    } catch (err: unknown) {
      if (isExecutionRevertedError(err)) {
        throw err;
      }

Reply:
It is still giving eslint issue if I remove ` as unknown`